File: puzzles/python3/temperatures/temperatures.py
from typing import List


def closest_to_zero(temperatures: List[int]):
    min_temperature = 5526
    for temperature in temperatures:
        if any([
            abs(temperature) < abs(min_temperature),
            abs(temperature) == abs(min_temperature) and temperature > min_temperature
        ]):
            min_temperature = temperature

    return 0 if len(temperatures) == 0 else min_temperature


if __name__ == "__main__":
    # the number of temperatures to analyse
    n = int(input())
    temperatures = list(map(int, input().split()))
    print(closest_to_zero(temperatures))

File: puzzles/swift/temperatures/temperatures.swift
import Glibc
import Foundation

public struct StderrOutputStream: TextOutputStream {
    public mutating func write(_ string: String) { fputs(string, stderr) }
}
public var errStream = StderrOutputStream()

let n = Int(readLine()!)! // the number of temperatures to analyse
var minTemperature = 5526

for i in ((readLine()!).split(separator: " ").map(String.init)) {
    let temperature = Int(i)!
    if (abs(temperature) < abs(minTemperature)) || (abs(temperature) == abs(minTemperature) && temperature > minTemperature) {
        minTemperature = temperature
    }
}

if n == 0 {
    print(0)
} else {
    print(minTemperature)
}